### PR TITLE
Bug 478530 - Add option to disable-enable build folder filtering

### PR DESCRIPTION
**Pull request by SimonScholz**
This PR contains 1 filter from the TAPI and 3 static filters different kinds of filters for the Package Explorer and the Project Explorer:

* Resource Filter from the TAPI
 * Acts like before, but with the new filter mechanism (activated by default)
* 3 static filters, which are supposed to be provided by the TAPI
 * Filter for the "build" folder (activated by default)
 * Filter for the ".gradle" folder (activated by default)
 * Filter for the "gradle" folder (turned off by default)

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=478530

File: org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/util/predicate/Predicates.java [hunk 1]
```diff
@@ -25,6 +34,74 @@
...
+
+    public static Predicate<IResource> isChildResourcesVisible() {
+        return isChildResourcesVisible;
+    }
+
+    public static Predicate<IResource> isBuildFolderVisible() {
+        return isBuildFolderVisible;
+    }
+
+    public static Predicate<IResource> isGraldeFolderVisible() {
```
Comment (Ian-Stewart-Binks) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/util/predicate/Predicates.java]: Minor typo in `Gralde`.

Comment (SimonScholz) [inline org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/util/predicate/Predicates.java]: Thanks @Ian-Stewart-Binks I just removed the typo

**Comment by donat** [comment]
The PR doesn't work correctly: if the ui plugin is not activated then the menu items don't show up.

**Comment by mwnorman** [comment]
Any idea when this fix is going to be merged?

**Comment by donat** [comment]
This has been superseded by the following changes on the master branch: https://github.com/eclipse/buildship/compare/f933de3c9404...c35d93ceda3a

**Pull request closed without merging**